### PR TITLE
RFC: use fill to print arrays with numbers & short strings

### PR DESCRIPTION
**Pull request by brodybits**
## Description

__updated__

Based on PR #10106, implements <https://github.com/prettier/prettier/issues/4210#issuecomment-376500821> for both numbers and strings up to 8 characters.

__This includes some quick testing.__

I hope this can help @breck7 as requested in <https://github.com/prettier/prettier/issues/4210#issuecomment-765584985>.

TODO:

- [x] update affected test snapshots
- [ ] resolve issue with cursorOffset & full-width characters in `tests_integration/__tests__/cursor-offset.js`
- [ ] add changelog

## Checklist

- [x] I’ve added tests to confirm my change works.
- ~~(If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).~~
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/print/array.js [hunk 1]
```diff
@@ -75,11 +78,25 @@ function printArray(path, options, print) {
         return element[itemsKey] && element[itemsKey].length > 1;
       });
 
+    const shouldUseFill =
+      n.elements.length > 1 &&
+      n.elements.every(
+        (element) =>
+          element &&
+          (isNumericLiteral(element) ||
+            (isStringLiteral(element) && element.value.length < 9))
```
Comment (fisker) [inline src/language-js/print/array.js:87]: Should  use  `raw` instead of `value`

Comment (fisker) [reply src/language-js/print/array.js:87]: And 8 is  not  enough for color `#RRGGBBAA`

Comment (brodybits) [reply src/language-js/print/array.js:87]: Using `raw` caused an error with `espree`. And this change did not work on `estree` or `meriyah`:

```suggestion
            (isStringLiteral(element) && (element.raw || element.value).length < 9))
```

It also did not resolve my issue with cursorOffset & full-width characters in `tests_integration/__tests__/cursor-offset.js`.

Comment (fisker) [reply src/language-js/print/array.js:86]: I mean https://github.com/prettier/prettier/blob/33078c61eb7388afa80a5057968cd656f7b7f0c8/src/language-js/utils.js#L968

**Comment by fisker** [comment]
Why did you close this?

**Comment by brodybits** [comment]
I felt as if I was going in a different direction from PR #10106, which seems to be the "accepted solution" ref: https://github.com/prettier/prettier/issues/4210#issuecomment-765670858

No problem if you think we should reopen this, but I think it should be considered in followup to PR #10106.

**Pull request closed without merging**